velox/exec/benchmarks/RowContainerSortBenchmark.cpp

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "velox/exec/RowContainer.h"
+#include "velox/external/timsort/TimSort.hpp"
+#include "velox/vector/DecodedVector.h"
+#include "velox/vector/SimpleVector.h"
+#include "velox/vector/tests/VectorTestUtils.h"
+#include "velox/vector/tests/utils/VectorMaker.h"
+
+namespace facebook::velox::test {
+namespace {
+
+template <typename T>
+VectorGeneratedData<T>
+testData(size_t length, size_t cardinality, bool sequences, size_t iteration) {
+  int32_t seqCount =
+      sequences ? std::max(2, (int32_t)(length / cardinality)) : 0;
+  int32_t seqLength = sequences ? std::max(2, (int32_t)(length / seqCount)) : 0;
+  return genTestDataWithSequences<T>(
```

Comment:
One more thing here, `genTestDataWithSequences` is generating same value sequences, so even when you disable the sequence, it can still form a monotonic increasing/decreasing sequence.  Can you replace this with something that we can have more control over the monotonic increasing/decreasing sequence length?

Reply:
@Yuhta I subsampled 500 rows data from the set that I get `runLength = 1.00947`. The perf is:
<img width="463" alt="image" src="https://github.com/facebookincubator/velox/assets/11849056/c0903a95-4f9a-4b9c-9539-f11cbfc2e292">

One open is that I should remove the previous benchmark leveraging `genTestDataWithSequences `, how do you think?

Reply:
Can you sample 50k rows?
For the sequence I think you can just use `genTestData` and just benchmark the uniform distribution case.  So we will have the parquet file to represent real world data and `genTestData` for worst case random data and I think that's enough for now.

Reply:
@Yuhta PR updated as you suggest. Please help take a review again and approve it if there is no further problem. Thanks.

Reply:
@Yuhta Switched to `genTestData` and remove redundant benchmark since now only `cardinality` matters. The latest data is:
<img width="464" alt="image" src="https://github.com/facebookincubator/velox/assets/11849056/933c1c52-26b6-4181-b006-2ebb33fddc24">
FYI.